images/scripts/create-anaconda-dnf-payload

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+# This file is part of Cockpit.
+#
+# Copyright (C) 2025 Red Hat, Inc.
+#
+# Licensed under the GNU Lesser General Public License v2.1 or later.
+
+# create-anaconda-dnf-payload  -- Create a DNF payload bundle for anaconda tests.
+# The output is a directory containing:
+#   repo/ (seeded for the Fedora version)
+
+import argparse
+import os
+import subprocess
+import xml.etree.ElementTree as ET
+
+from lib.constants import BOTS_DIR
+from machine import testvm
+
+
+def run(cmd: list[str]) -> None:
+    subprocess.check_call(cmd)
+
+
+def parse_required_packages_from_comps(comps_path: str) -> str:
+    """Return a space-separated list of required package names from a comps.xml file.
+
+    Considers only mandatory and default entries.
+    """
+    names: set[str] = set()
+    wanted = {"core", "critical-path-base", "anaconda-tools"}
+    root = ET.parse(comps_path).getroot()
+    for group in root.findall("group"):
+        group_id = (group.findtext("id") or "").strip()
+        if group_id not in wanted:
+            continue
+        pkg_list = group.find("packagelist")
+        if pkg_list is None:
+            continue
+        for req in pkg_list.findall("packagereq"):
+            req_type = req.get("type", "")
+            if req_type in ("mandatory", "default") and req.text:
+                names.add(req.text.strip())
+    return " ".join(sorted(names))
+
+
+def download_packages_on_guest(machine: testvm.VirtMachine, baseurl: str, destdir: str, packages: str) -> None:
+    """Download the given packages and dependencies into destdir using dnf5 on the guest."""
+    if not packages:
+        return
+    machine.execute(
+        f"dnf5 -y --disablerepo='*' --repofrompath=base,{baseurl} --enablerepo=base --forcearch=x86_64 "
+        f"download --resolve --alldeps --destdir '{destdir}' {packages}",
+        timeout=3600,
+    )
+
+
+def build_dnf_payload(image: str, output: str) -> None:
+    fedora_version = image.split('-')[1]
+
+    run([os.path.join(BOTS_DIR, "image-download"), image])
+    machine = testvm.VirtMachine(image=image, memory_mb=4096)
+    try:
+        machine.start()
+        machine.wait_boot()
+        machine.execute("dnf install -y dnf5-plugins createrepo_c python3 rsync curl", timeout=300)
```

Comment:
This mixes `dnf` and `dnf5`. Intentional?

Reply:
dnf executable defaults to dnf5 since few fedora versions.

Reply:
Right :) I think most of these dependencies are actually not needed anymore, after some cleanups. Trying it out with keeping only createrepo_c which I still use